src/shogun/kernel/SigmoidKernel.cpp

```diff
@@ -52,9 +53,26 @@ bool CSigmoidKernel::init(CFeatures* l, CFeatures* r)
 
 void CSigmoidKernel::init()
 {
-	gamma=0.0;
-	coef0=0.0;
+	gamma = 0.0;
+	coef0 = 0.0;
 
-	SG_ADD(&gamma, "gamma", "Gamma.", ParameterProperties::HYPER);
+	SG_ADD(
+	    &gamma, "gamma", "Scaler for the dot product.",
+	    ParameterProperties::HYPER | ParameterProperties::AUTO, [this]() {
```

Comment:
we need to think about a way to document what "auto" here means. In particular, I think the strategy/function that computes the default value should come with a description string that can be queried by the user. Any ideas for that?

Reply:
Well the description can be access with `AnyParameterProperties::get_description()`? And then can be queried from the interface via `SGObjet.parameter_description(name)`. Just need to improve the string description here.

Reply:
I mean a description for how the auto is computed. I think the description should live at the same place as the function that computed it so It can also be queries